client.go

```diff
@@ -299,9 +340,23 @@ func (r *marathonClient) apiCall(method, path string, body, result interface{})
 
 		// step: check for a successfull response
 		if response.StatusCode >= 200 && response.StatusCode <= 299 {
+
 			if result != nil {
-				if err := json.Unmarshal(respBody, result); err != nil {
-					return fmt.Errorf("failed to unmarshal response from Marathon: %s", err)
+				// If we have a deployment ID header and no response body, give them that
+				deploymentID := response.Header.Get(deploymentHeader)
+				if len(respBody) == 0 && deploymentID != "" {
```

Comment:
That's true for PUT, but DELETE only returns the deployment ID in the header. As far as I can tell, that's the only flow in which this occurs, since otherwise the pod is contained in the body. Since in other situations we have other ways of determining the status of the deployment, I'm not too concerned about losing the deployment ID to the pod body, but I'm open to feedback there.